packages/router-devtools-core/src/BaseTanStackRouterDevtoolsPanel.tsx

```diff
@@ -126,6 +149,45 @@ function RouteComp({
     }
   })
 
+  const navigationTarget = createMemo<string | undefined>(() => {
+    if (isRoot) return undefined // rootRouteId has no path
+    if (!route.path) return undefined // no path to navigate to
+
+    // flatten all params in the router state, into a single object
+    const allParams = matches()
+      .flatMap((m) => m.params)
+      .reduce((prev, curr) => {
+        const keys = Object.keys(curr)
+        for (const key of keys) {
+          if (prev[key] === undefined) {
+            prev[key] = curr[key]
+          }
+        }
+        return prev
+      }, {})
```

Comment:
```suggestion
    const allParams = Object.assign({}, ...matches().reverse().flatMap(m => m.params))
```

@SeanCassiere how about this apporach?

Reply:
Had to make a few modifications, but the essential gist of your suggestion has been applied.